core/src/main/scala/tech/beshu/ror/api/AuthMockApi.scala

```diff
@@ -16,168 +16,354 @@
 import tech.beshu.ror.accesscontrol.blocks.definitions.ldap.LdapService
-import tech.beshu.ror.accesscontrol.blocks.definitions.{ExternalAuthenticationService, ExternalAuthorizationService}
-import tech.beshu.ror.accesscontrol.blocks.mocks.MocksProvider.ExternalAuthenticationServiceMock.ExternalAuthenticationUserMock
-import tech.beshu.ror.accesscontrol.blocks.mocks.MocksProvider.ExternalAuthorizationServiceMock.ExternalAuthorizationServiceUserMock
-import tech.beshu.ror.accesscontrol.blocks.mocks.MocksProvider.LdapServiceMock.LdapUserMock
+import tech.beshu.ror.accesscontrol.blocks.definitions.{ExternalAuthenticationService => AuthenticationService, ExternalAuthorizationService => AuthorizationService}
 import tech.beshu.ror.accesscontrol.blocks.mocks.MocksProvider.{ExternalAuthenticationServiceMock, ExternalAuthorizationServiceMock, LdapServiceMock}
-import tech.beshu.ror.accesscontrol.blocks.mocks.{MapsBasedMocksProvider, MutableMocksProviderWithCachePerRequest}
+import tech.beshu.ror.accesscontrol.blocks.mocks.{MapsBasedMocksProvider, MocksProvider, MutableMocksProviderWithCachePerRequest}
 import tech.beshu.ror.accesscontrol.domain.{Group, User}
-import tech.beshu.ror.boot.RorSchedulers
-import tech.beshu.ror.utils.HttpOps
-import tech.beshu.ror.{Constants, RequestId}
+import tech.beshu.ror.api.AuthMockApi.AuthMockResponse.{Failure, ProvideAuthMock, UpdateAuthMock}
+import tech.beshu.ror.api.AuthMockApi.AuthMockService._
+import tech.beshu.ror.boot.RorInstance.TestEngineServices
+import tech.beshu.ror.boot.{RorInstance, RorSchedulers}
+import tech.beshu.ror.configuration.RorConfig
+import tech.beshu.ror.utils.CirceOps.CirceErrorOps
 
-import scala.concurrent.duration._
 import scala.language.postfixOps
 
-class AuthMockApi(mockProvider: MutableMocksProviderWithCachePerRequest)
+class AuthMockApi(rorInstance: RorInstance,
+                  mockProvider: MutableMocksProviderWithCachePerRequest)
   extends Logging {
 
+  import AuthMockApi.Utils._
+  import AuthMockApi.Utils.decoders._
   import AuthMockApi._
-  import AuthMockApi.coders._
 
   def call(request: AuthMockRequest)
           (implicit requestId: RequestId): Task[AuthMockResponse] = {
-    val apiCallResult = request.aType match {
-      case AuthMockRequest.Type.UpdateAuthMock => updateAuthMock(request.body, authMockTtlFrom(request.headers))
-      case AuthMockRequest.Type.InvalidateAuthMock => invalidateAuthMock()
+    val response = request.aType match {
+      case AuthMockRequest.Type.ProvideAuthMock => provideAuthMock()
+      case AuthMockRequest.Type.UpdateAuthMock => updateAuthMock(request.body)
     }
-    apiCallResult
+    response
       .executeOn(RorSchedulers.restApiScheduler)
   }
 
-  private def updateAuthMock(body: String,
-                             authMockTtl: Option[FiniteDuration])
+  private def provideAuthMock()
+                             (implicit requestId: RequestId): Task[AuthMockResponse] = {
+    rorInstance.currentServices().map {
```

Comment:
I'd rather encapsulate services inside a RorConfig class (not the raw one we have, but a substitute of ROR yaml settings represented in scala code - I've mentioned once about it /it was rather a mistake to parse yaml to core directly - it complicated many things/). Inside this class, we would have only the services obviously (and maybe a reference to RawRorConfig). RorInstance should expose the current RorConfig, not just the services.